traitlets/config/application.py

```diff
@@ -464,13 +466,16 @@ def initialize_subcommand(self, subc, argv=None):
         """Initialize a subcommand with argv."""
         subapp,help = self.subcommands.get(subc)
 
-        if isinstance(subapp, six.string_types):
-            subapp = import_item(subapp)
+        if callable(subapp):
+            self.subapp = subapp(self)
```

Comment:
Indeed!

Would this be ok?

    if callable(subapp) and not isinstance(subapp, Application):
        ...